File: app/src/main/java/com/dewonderstruck/apps/ashx0/binding/FragmentDataBindingComponent.java
package com.dewonderstruck.apps.ashx0.binding;

import androidx.databinding.DataBindingComponent;
import androidx.fragment.app.Fragment;

/**
 * A Data Binding Component implementation for fragments.
 */
public class FragmentDataBindingComponent implements DataBindingComponent {
    private final FragmentBindingAdapters adapter;

    public FragmentDataBindingComponent(Fragment fragment) {
        this.adapter = new FragmentBindingAdapters(fragment);
    }

    @Override
    public FragmentBindingAdapters getFragmentBindingAdapters() {
        return adapter;
    }
}

File: app/src/main/java/com/dewonderstruck/apps/ashx0/binding/FragmentDataBindingComponent.kt
package com.dewonderstruck.apps.ashx0.binding

import androidx.databinding.DataBindingComponent
import androidx.fragment.app.Fragment

/**
 * A Data Binding Component implementation for fragments.
 */
class FragmentDataBindingComponent(fragment: Fragment?) : DataBindingComponent {
    private val adapter: FragmentBindingAdapters
    override fun getFragmentBindingAdapters(): FragmentBindingAdapters {
        return adapter
    }

    init {
        adapter = FragmentBindingAdapters(fragment)
    }
}